app/routes/_auth.signup.tsx

```diff
@@ -66,20 +93,75 @@ export default function Component() {
         <Card>
           <Form method="post" className="w-full flex flex-col items-center justify-center gap-5">
             <h1 className="text-4xl">Sign Up</h1>
-            <Input name="email" type="email" autoComplete="email" placeholder="Email" label="Email" required />
-            <Input name="name" autoComplete="username" type="text" placeholder="Name" label="Name" required />
-            <Input name="password" type="password" placeholder="Password" label="Password" required />
+
+            <Input
+              defaultValue={actionData?.fields?.name}
+              name="name"
+              autoComplete="name"
+              type="text"
+              placeholder="Name"
+              label="Name"
+              required
+              aria-invalid={Boolean(actionData?.fieldErrors?.name)}
+              aria-errormessage={actionData?.fieldErrors?.name ? 'name-error' : undefined}
+            />
+            {actionData?.fieldErrors?.name && (
+              <p className="text-sm text-red-600" id="name-error" role="alert">
+                {actionData.fieldErrors.name._errors[0]}
+              </p>
+            )}
```

Comment:
The styling and presentation of these error messages can be improved for better UX. We could also probably make the error into a reusable component.